ironfish/src/rpc/routes/wallet/getBalances.test.ts

```diff
@@ -85,18 +85,137 @@ describe('Route wallet/getBalances', () => {
+          assetOwner: Buffer.from('Iron Fish', 'utf8'),
+          confirmed: BigInt(2000000000),
+          unconfirmed: BigInt(2000000000),
+          pending: BigInt(2000000000),
+          available: BigInt(2000000000),
+          unconfirmedCount: 0,
+          pendingCount: 0,
+          blockHash: null,
+          sequence: null,
+        },
+      ]
+
+      const getBalances = jest
+        .spyOn(account, 'getBalances')
+        // eslint-disable-next-line @typescript-eslint/require-await
+        .mockImplementationOnce(async function* (_confirmations) {
+          for (const balance of mockBalances) {
+            yield balance
+          }
+        })
+
+      jest.spyOn(account, 'getAsset').mockReturnValueOnce(
+        Promise.resolve({
+          id: asset.id(),
+          metadata: asset.metadata(),
+          name: asset.name(),
+          nonce: asset.nonce(),
+          owner: asset.owner(),
+          createdTransactionHash: Buffer.alloc(32),
+          blockHash: Buffer.alloc(32),
+          sequence: null,
+          supply: null,
+        }),
       )
+
+      const verifyAsset = jest
+        .spyOn(node.assetsVerifier, 'verify')
+        .mockImplementation((assetId) => {
+          if (!(typeof assetId === 'string')) {
+            assetId = assetId.toString('hex')
+          }
+          if (assetId === Asset.nativeId().toString('hex')) {
+            return { status: 'verified' }
+          } else {
+            return { status: 'unverified' }
+          }
+        })
+
+      const response = await routeTest.client.wallet.getAccountBalances({
+        account: account.name,
+      })
+
+      expect(getBalances).toHaveBeenCalledTimes(1)
+      expect(verifyAsset).toHaveBeenCalledWith(Asset.nativeId())
+      expect(verifyAsset).toHaveBeenCalledWith(asset.id())
+
+      expect(response.content.balances).toEqual([
+        {
+          ...mockBalances[0],
+          assetId: mockBalances[0].assetId.toString('hex'),
+          assetName: mockBalances[0].assetName.toString('hex'),
+          assetOwner: mockBalances[0].assetOwner.toString('hex'),
+          assetVerification: { status: 'unverified' },
+          confirmed: mockBalances[0].confirmed.toString(),
+          unconfirmed: mockBalances[0].unconfirmed.toString(),
+          pending: mockBalances[0].pending.toString(),
+          available: mockBalances[0].available.toString(),
+        },
+        {
+          ...mockBalances[1],
+          assetId: mockBalances[1].assetId.toString('hex'),
+          assetName: mockBalances[1].assetName.toString('hex'),
+          assetOwner: mockBalances[1].assetOwner.toString('hex'),
+          assetVerification: { status: 'verified' },
+          confirmed: mockBalances[1].confirmed.toString(),
+          unconfirmed: mockBalances[1].unconfirmed.toString(),
+          pending: mockBalances[1].pending.toString(),
+          available: mockBalances[1].available.toString(),
+        },
+      ])
```

Comment:
This is a lot of code for just testing asset verification, I think this actually can be reduced down to a few lines of code. I'm willing to pair on this to go through this test. It also duplicates the verification logic through mocking in a strange way.

Reply:
Reduced the test code, and simplified the mock. I can also put the test preparation code in a common function to reduce the number of lines even further if you want.

Reply:
I think this is good for now. If I have any other recommendations I can just show it in a proposal patch later.